test/unit/libraries/EmergencyProtection.t.sol

```diff
@@ -1,352 +1,223 @@
-        assertEq(_emergencyProtection.emergencyProtectionEndsAfter, newProtectionDuration.addTo(Timestamps.now()));
-        assertEq(_emergencyProtection.emergencyModeDuration, newEmergencyModeDuration);
-        assertEq(_emergencyProtection.emergencyModeEndsAfter, Timestamps.ZERO);
+        assertTrue(EmergencyProtection.isEmergencyModeActive(ctx));
+        assertEq(Timestamp.unwrap(ctx.emergencyModeEndsAfter), block.timestamp + 3600);
     }
 
-    function test_setup_same_execution_committee() external {
-        Duration protectionDuration = Durations.from(100 seconds);
-        Duration emergencyModeDuration = Durations.from(100 seconds);
-        address executionCommittee = makeAddr("executionCommittee");
-
-        _setup(_emergencyGovernance, address(0x1), executionCommittee, protectionDuration, emergencyModeDuration);
-
-        Duration newProtectionDuration = Durations.from(200 seconds);
-        Duration newEmergencyModeDuration = Durations.from(300 seconds);
-
-        vm.expectEmit();
-        emit EmergencyProtection.EmergencyActivationCommitteeSet(address(0x2));
-        vm.expectEmit();
-        emit EmergencyProtection.EmergencyProtectionEndDateSet(newProtectionDuration.addTo(Timestamps.now()));
-        vm.expectEmit();
-        emit EmergencyProtection.EmergencyModeDurationSet(newEmergencyModeDuration);
-
-        vm.recordLogs();
-        _setup(_emergencyGovernance, address(0x2), executionCommittee, newProtectionDuration, newEmergencyModeDuration);
+    function test_ActivateEmergencyMode_RevertOn_ProtectionExpired() external {
+        Duration untilExpiration =
+            Durations.between(ctx.emergencyProtectionEndsAfter, Timestamps.from(block.timestamp)).plusSeconds(1);
 
-        Vm.Log[] memory entries = vm.getRecordedLogs();
-        assertEq(entries.length, 3);
+        _wait(untilExpiration);
 
-        assertEq(_emergencyProtection.emergencyActivationCommittee, address(0x2));
-        assertEq(_emergencyProtection.emergencyExecutionCommittee, executionCommittee);
-        assertEq(_emergencyProtection.emergencyProtectionEndsAfter, newProtectionDuration.addTo(Timestamps.now()));
-        assertEq(_emergencyProtection.emergencyModeDuration, newEmergencyModeDuration);
-        assertEq(_emergencyProtection.emergencyModeEndsAfter, Timestamps.ZERO);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                EmergencyProtection.EmergencyProtectionExpired.selector, ctx.emergencyProtectionEndsAfter
+            )
+        );
+        EmergencyProtection.activateEmergencyMode(ctx);
     }
 
-    function test_setup_same_protected_till() external {
-        Duration protectionDuration = Durations.from(100 seconds);
-        Duration emergencyModeDuration = Durations.from(100 seconds);
-
-        _setup(_emergencyGovernance, address(0x1), address(0x2), protectionDuration, emergencyModeDuration);
+    function test_DeactivateEmergencyMode() external {
```

Comment:
Use mixed case with the first letter lowercase for method names, e.g., `test_deactivateEmergencyMode()`. Additionally, it's recommended to append the suffix `_HappyPath_[SpecificScenarioIfApplicable]` to tests that validate correct/default behavior, such as `test_deactivateEmergencyMode_HappyPath()`.